### PR TITLE
Draw support QJIT qnodes

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**

Catalyst users cannot draw a circuit.

**Description of the Change:**

- Add drawing capacities in PennyLane for Catalyst
- The ZX test suite is changed to external libraries test suite.

**Benefits:**

Catalyst users can draw QJIT circuits.

### CODE REVIEW

**Review by mudit2812 — COMMENTED** [review]
Looks good! Can you add an entry to the improvements section of the changelog? I'm ready to approve otherwise.

**Comment by dime10** [comment]
[sc-41308]

File: pennylane/drawer/draw.py
Comment (maliasadi) [inline pennylane/drawer/draw.py:38]:
```suggestion
    try:
        pkg_resources.get_distribution("pennylane_catalyst")
        return isinstance(qnode, catalyst.QJIT)
    except pkg_resources.DistributionNotFound:
        return False
```

**Review by maliasadi — COMMENTED** [review]
Thanks @rmoyard! I'll approve after merging the suggestion and checking tests :tada:

File: pennylane/drawer/draw.py
Comment (rmoyard) [reply pennylane/drawer/draw.py:1]: Yes I will add a few lines

File: pennylane/drawer/draw.py
Comment (rmoyard) [reply pennylane/drawer/draw.py]: Thanks let me try that!

**Review by mudit2812 — APPROVED** [review]

**Review by maliasadi — APPROVED** [review]

File: doc/releases/changelog-dev.md
Comment (trbromley) [inline doc/releases/changelog-dev.md:11]:
```suggestion
  import catalyst
```

**Pull request merged**
